src/modules/SX126x/SX126x.cpp

```diff
@@ -443,7 +443,91 @@ int16_t SX126x::startTransmit(uint8_t* data, size_t len, uint8_t addr) {
 }
 
 int16_t SX126x::startReceive(uint32_t timeout) {
-  // set DIO mapping
+  int16_t state = startReceiveCommon();
+  if(state != ERR_NONE) {
+    return(state);
+  }
+
+  // set mode to receive
+  state = setRx(timeout);
+
+  return(state);
+}
+
+int16_t SX126x::startReceiveDutyCycle(uint32_t rxPeriod_us, uint32_t sleepPeriod_us)
+{
+  // datasheet claims time to go to sleep is ~500us, same to wake up.
+  // compensate for that 1ms + tcxo delay:
+  uint32_t transitionTime = _tcxoDelay_us + 1000;
+  if (sleepPeriod_us < transitionTime + 16) {
+    return(ERR_INVALID_SLEEP_PERIOD);
+  }
+  sleepPeriod_us -= transitionTime;
+
+  // divide by 15.625
+  uint32_t rxPeriodRaw = (rxPeriod_us * 8) / 125;
+  uint32_t sleepPeriodRaw = (sleepPeriod_us * 8) / 125;
+
+  // 24 bit limit:
+  if ((rxPeriodRaw & 0xFF000000) || (sleepPeriodRaw & 0xFF000000)) {
```

Comment:
This is a bit too generic - caller doesn't know if the error here is caused by `rxPeriod` or `sleepPeriod`. It might be a better idea to check them independently and throw specific errors (e.g. ERR_SLEEP_PERIOD_TOO_LONG and ERR_RX_PERIOD_TOO_LONG).

Reply:
Would it be OK to just have `ERR_INVALID_SLEEP_PERIOD` and `ERR_INVALID_RX_PERIOD`? In most other radiolib functions there aren't distinct errors for too large and too small.

Reply:
That's true actually, let's not compilcate things. ERR_INVALID_SLEEP_PERIOD and ERR_INVALID_RX_PERIOD it is.